vsce compare changes in dev

mirnawong1 opened the pull request:
this pr adds a page and snippet to describe the compare changes in local dev feature for vsce. 

[X ] needs technical review
[X] needs pm review
[X] needs docs review

[notion project](https://www.notion.so/dbtlabs/Compare-changes-in-VSCE-287bb38ebda780c1bac8caaf9adfcee7?source=copy_link)

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


- https://docs-getdbt-com-git-vsce-compare-changes-dbt-labs.vercel.app/docs/about-dbt-extension
- https://docs-getdbt-com-git-vsce-compare-changes-dbt-labs.vercel.app/docs/dbt-extension-features
- https://docs-getdbt-com-git-vsce-compare-changes-dbt-labs.vercel.app/docs/deploy/advanced-ci
- https://docs-getdbt-com-git-vsce-compare-changes-dbt-labs.vercel.app/docs/fusion/vs-compare-changes

<!-- end-vercel-deployment-preview -->

mirnawong1 committed:
add new files

mirnawong1 committed:
add beta

mirnawong1 committed:
add tweaks

mirnawong1 committed:
update table

mirnawong1 committed:
update step

mirnawong1 committed:
update greg comments

mirnawong1 committed:
merge conflict

mirnawong1 committed:
updates

mirnawong1 committed:
update compare changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 commented on website/docs/docs/deploy/advanced-ci.md:
```diff
@@ -23,14 +26,42 @@ You can opt into Advanced CI in <Constant name="cloud" />. Please refer to [Acco
+
 For [CI jobs](/docs/deploy/ci-jobs) that have the [**dbt compare** option enabled](/docs/deploy/ci-jobs#set-up-ci-jobs), <Constant name="cloud" /> compares the changes between the last applied state of the production environment (defaulting to deferral for lower compute costs) and the latest changes from the pull request, whenever a pull request is opened or new commits are pushed.  
 
+You can also compare changes in development. For more details, see [Compare changes in development](#compare-changes-in-development).
```
```suggestion
You can also compare changes in development. For more details, see [Compare changes in local development](/docs/fusion/vs-compare-changes).
```

mirnawong1 committed:
Update website/docs/docs/deploy/advanced-ci.md

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+---
+title: "Compare changes in local development"
```
making it clear this is for local development

mirnawong1 replied:
we can either turn entire content or parts of content into snippet if we need to reuse

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+---
+title: "Compare changes in local development"
+id: "vs-compare-changes"
+description: "See how code edits will change your data while you develop in VS Code—without waiting on CI."
```
```suggestion
description: "Compare how code edits will change your data while you develop in locally with the dbt VS Code extension — without waiting on CI."
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+sidebar_label: "Compare changes"
+---
+
+# Compare changes in development <Lifecycle status="beta,managed,managed_plus" />
```
```suggestion
# Compare changes in local development <Lifecycle status="beta,managed,managed_plus" />
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 commented on website/snippets/_compare-changes-tldr.md:
```diff
@@ -0,0 +1,7 @@
+
+You can use compare changes in development, powered by the <Constant name="fusion_engine"/>, to compare your current working copy against your `manifest.json` (for example, your last production state) — directly in your editor.
+
+For more details on how to use this feature, see [Compare changes in development](/docs/fusion/vs-compare-changes).
```
```suggestion
For more details on how to use this feature, see [Compare changes in local development](/docs/fusion/vs-compare-changes).
```

mirnawong1 committed:
Update website/snippets/_compare-changes-tldr.md

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- You have a dbt Enterprise or Enterprise+ account.
+- You have a developer seat license.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/install-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
```
is this correct? or not needed?

connie-carey replied:
I'm not sure it's needed. @davidharting can you confirm?

davidharting replied:
This is correct! Today the "compare" entitlement is tied to Advanced CI for both platform and local VSCE. 

If we want to de-couple those in the feature we will need to adjust the entitelments. But for now this line is 100% accurate!

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- You have a developer seat license.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/install-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
+- Use a <Constant name="fusion" /> supported data platform: BigQuery, Databricks, Redshift, or Snowflake. Support for additional data platforms coming soon.
```
presume this is right since its fusion?

davidharting replied:
That is correct - compare is built to work in any supported platform (it defers to adapters to actually query the warehouse). 

So this list here should just be the fusion list!

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+- You have a dbt Enterprise or Enterprise+ account.
+- You have a developer seat license.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/install-fusion) in your editor.
```
```suggestion
- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
```

mirnawong1 committed:
Apply suggestion from @mirnawong1

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
fix links

mirnawong1 committed:
Update dbt VS Code extension documentation

Added support for VS Code, Cursor, and Windsurf to the dbt extension documentation.

mirnawong1 commented on website/sidebars.js:
```diff
@@ -398,6 +398,7 @@ const sidebarSettings = {
             "docs/about-dbt-extension",
             "docs/dbt-extension-features",
             "docs/install-dbt-extension",
+            "docs/fusion/vs-compare-changes",
```
```suggestion
```

mirnawong1 commented on website/sidebars.js:
```diff
@@ -398,6 +398,7 @@ const sidebarSettings = {
             "docs/dbt-extension-features",
             "docs/install-dbt-extension",
+            "docs/fusion/vs-compare-changes",
             "docs/configure-dbt-extension",
```
```suggestion
            "docs/configure-dbt-extension",
            "docs/fusion/vs-compare-changes",
```

mirnawong1 committed:
Apply suggestions from code review

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

connie-carey commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+The dbt VS Code extension, powered by the <Constant name="fusion_engine"/>, can preview and compare changes to your data caused by your local edits (like added/removed rows and so on) &mdash; directly in your editor and without waiting on CI. The dbt extension supports VS Code, Cursor, and Windsurf.
+
+Compare changes in development is [different to Advanced CI compare changes](#how-this-differs-from-advanced-ci) in that it allows you to:
+- Check impact early on amd catch changes before you open a PR or run a [CI job](/docs/deploy/ci-jobs) in deployment.
```
typo "amd" should be "and"

mirnawong1 replied:
🙃 great catch!

connie-carey commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+
+## Prerequisites
+
+- You have a dbt Enterprise or Enterprise+ account.
```
Should we call out that it must be connected via dbt_cloud.yml locally?

connie-carey commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+## Prerequisites
+
+- You have a dbt Enterprise or Enterprise+ account.
+- You have a developer seat license.
```
Is this important @davidharting? Are we checking for the user's platform user license or just account plan tier?

davidharting replied:
Correct - we do _not_ actually check that the user has a developer license for this feature or really any feature in VSCE!


---
Full details from my code walkthrough to verify this:


We defer to the Fusion licensing check to determine if the user has compare.

For a user to get the "compare" feature they must:

- Be in an account that is Enterprise / Enterprise+ (as noted above)
- Account is  __opted in to Advanced CI__ (as noted below)

mirnawong1 replied:
thanks team, removed!

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
+- Use a <Constant name="fusion" /> supported data platform: BigQuery, Databricks, Redshift, or Snowflake. Support for additional data platforms coming soon.
+- If you've configured [automatic deferral](/docs/cloud/about-cloud-develop-defer), you need a successful job run. To use compare changes manually without a successful job run, you can manually copy a `manifest.json` and specify the state directory.
```
Good and important callout!

Not sure if this is called out anywhere else, but you have to configure the state directory in the dbt vscode extension settings

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+To use compare changes in development, follow these steps:
+
+1. Open a SQL model file in your editor.
+2. Make sure to successfully build the model at least once with `dbt build`. This gives dbt a baseline for comparison and allows you to compare your changes against the last successful build of the model. If you don't have a successful build, compare changes won't work effectively.
```
Once @aricallen's https://github.com/dbt-labs/fs/pull/8875 is merged, hitting the Compare button will always run build before compare. 

Sorry I know it's hard to document this stuff when the behavior is always changing!

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+1. Open a SQL model file in your editor.
+2. Make sure to successfully build the model at least once with `dbt build`. This gives dbt a baseline for comparison and allows you to compare your changes against the last successful build of the model. If you don't have a successful build, compare changes won't work effectively.
+3. Make some changes to the model, like adding a new column or modifying an existing one (for example, `left_join_customers` to `right_join_customers`).
+4. Use the command palette and search for the [**dbt: Compare changes**](vscode://dbtLabsInc/dbt.compareModel) _or_ click the **Compare** tab in the editor toolbar.
```
I wanted to upload a screenshot here to get a chance to make sure I understand what you mean by compare tab and editor toolbar. VSCode's official language is so clunky!

<img width="3602" height="2272" alt="Image" src="https://github.com/user-attachments/assets/b1bbdfe1-0db9-4d4d-97a9-51b58be6d683" />

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+
+## How it works
+
+Compare changes in development works by comparing your current working copy against your `manifest.json` (for example, your last production state) — directly in your editor. It uses the same credentials as the CI job, as defined in the CI job’s environment.
```
It could be useful to clarify that it compares the _materialized_ model in your dev schema (your dev schema is determined by whatever schema your active profile is using) to the materialized model pointed to in manifest.json. 

Maybe this won't be confusing for our readers, but as a non-dbt person, I always get confused on this topic, because it's not clear to me when dbt cares about the sql contents of my file versus what's in the warehouse.

In this case, it's _always_ the warehouse - we compare the built models to each other across dev and whatever state is represented by the other manifest.json.

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+Compare changes in development works by comparing your current working copy against your `manifest.json` (for example, your last production state) — directly in your editor. It uses the same credentials as the CI job, as defined in the CI job’s environment.
+
+  - If you're using <Constant name="dbt_platform" />'s deferral (recommended): Yes. You need at least one successful job run in the environment you are deferring to (usually production). This allows Fusion to auto-download the deferred manifest and use that as your baseline state.
+  - If you're manually setting a state directory: No. You can manually point the extension to a manifest.json (for example, copied from another environment) without needing a job run.run.
```
Thanks for being careful to always call out the differences between using your own manifest.json versus automatic deferral!

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+
+
+<Expandable alt_header="Are queries run on behalf of the developer?"> 
+  Yes. All comparison queries in development are run using your local development credentials, directly from the dbt VS Code extension. The <Constant name="fusion_engine" /> uses your credentials to execute comparison queries in your warehouse. The results are stored locally by the extension for caching and faster repeated comparisons.
```
> The results are stored locally by the extension for caching and faster repeated comparisons.

The results are only stored in memory, so that we can keep them populated into the Compare tab for that file until you re-run.

Unfortunately, the caching does _not_ speed up repeated comparisons 🐢

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+<Expandable alt_header="Do I need to run `dbt build` every time I make a change?"> 
+  You don’t need to run `dbt build` after every change, but you do need at least one successful build per model to establish the baseline.
+  - You need one successful `build` of the model to create a baseline table for comparison.
+  - After that, you can edit your model and run **Compare** without rebuilding every time.
```
_Today_ you do need to rebuild the model after every change for Compare to pick up the change. But Aric's PR i linked earlier will fix that.

davidharting commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,91 @@
+  You don’t need to run `dbt build` after every change, but you do need at least one successful build per model to establish the baseline.
+  - You need one successful `build` of the model to create a baseline table for comparison.
+  - After that, you can edit your model and run **Compare** without rebuilding every time.
+  - If your model becomes invalid (for example, errors or breaking schema changes), you’ll need to rebuild to create a new valid baseline before comparing again.
```
This isn't quite how I would word it. I would reserve the word "baseline" for the deferral environment (either auto deferral or state directory manifest.json).

If you have an invalid model, then it cannot be built, and thus we cannot do a comparison. So that aspect of it rings true!

davidharting reviewed (approved):
First batch of comments! This pretty much looks ready to go with a few caveats so going ahead and approving now, but happy to re-review later =D

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
feedback

mirnawong1 committed:
add why

mirnawong1 committed:
add bullets

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 commented:
thanks so much @davidharting @connie-carey ! i've rolled in those changes now

mirnawong1 committed:
fix img

mirnawong1 committed:
Merge branch 'vsce-compare-changes' of github.com:dbt-labs/docs.getdbt.com into vsce-compare-changes

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

matthewshaver commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -12,7 +12,9 @@ The <Constant name="dbt_platform" /> natively supports developing using a comman
 :::note CLI compatibility
 
-The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). It is not compatible with existing installations of the <Constant name="core" /> or <Constant name="fusion_engine" /> CLIs.
+The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). If your dbt Cloud project uses Fusion as its engine, the platform CLI will automatically use Fusion for remote execution — no separate Fusion install is required.
```
Hey @mirnawong1 had a separate PR for this section that we were workshopping [here](https://github.com/dbt-labs/docs.getdbt.com/pull/8866)

I think it might be beneficial, rather than opening the page with a long callout, let's move all of this info down to the prerequisites and explain it there in detail. That would provide a more consistent experience. What do you think of that?

mirnawong1 replied:
oh why is this here? i don't remember even adding this or maybe i was drafting it before i chatted with you. @matthewshaver should i just remove this since you're tackling this? i don't want to mix different changes into one pr

mirnawong1 replied:
removing!

matthewshaver commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -12,7 +12,9 @@ The <Constant name="dbt_platform" /> natively supports developing using a comman
 :::note CLI compatibility
 
-The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). It is not compatible with existing installations of the <Constant name="core" /> or <Constant name="fusion_engine" /> CLIs.
+The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). If your dbt Cloud project uses Fusion as its engine, the platform CLI will automatically use Fusion for remote execution — no separate Fusion install is required.
```
```suggestion
The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). 

If your <Constant name="dbt_platform" /> project uses <Constant name="Fusion" /> as its engine, the platform CLI will automatically use it for remote execution, so no local <Constant name="Fusion" /> installation is required.
```

matthewshaver commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -12,7 +12,9 @@ The <Constant name="dbt_platform" /> natively supports developing using a comman
-The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). It is not compatible with existing installations of the <Constant name="core" /> or <Constant name="fusion_engine" /> CLIs.
+The dbt CLI is a <Constant name="dbt_platform" /> tool available to users on any [available plan](https://www.getdbt.com/pricing). If your dbt Cloud project uses Fusion as its engine, the platform CLI will automatically use Fusion for remote execution — no separate Fusion install is required.
+
+However, if you also have <Constant name="core" /> or the <Constant name="fusion_engine" /> CLI installed locally, note that all three tools use the `dbt` command. You'll need to configure aliases or use explicit binary paths to invoke the right tool. See the [FAQs](#faqs) below for guidance.
```
```suggestion
If you have <Constant name="core" /> and/or the <Constant name="fusion_engine" /> CLI installed locally, all three tools use the `dbt` command. You'll need to configure aliases or use explicit binary paths to invoke the right tool. See the [FAQs](#faqs) below for guidance.
```

matthewshaver commented on website/docs/docs/deploy/advanced-ci.md:
```diff
@@ -23,14 +26,42 @@ You can opt into Advanced CI in <Constant name="dbt" />. Please refer to [Accoun
 
 ## Compare changes feature {#compare-changes}
 
+:::tip Compare changes in development
```
This is a lengthy tip - It could be useful as an H3 in this section and make it easier to link to if useful elsewhere

mirnawong1 replied:
yea i did think of that too but its such a. long page so was trying to 'shorten' it with a callout. but yea will switch from tip to an h4/5 or something

matthewshaver commented on website/docs/docs/deploy/advanced-ci.md:
```diff
@@ -23,14 +26,42 @@ You can opt into Advanced CI in <Constant name="dbt" />. Please refer to [Accoun
 
+:::tip Compare changes in development
+
+You can also use compare changes in development to preview changes to your data caused by your local edits (like added/removed rows and so on) &mdash; directly in your editor and without waiting on CI. Compare changes in development is available through the dbt VS Code extension, which is powered by the <Constant name="fusion_engine"/>.
```
```suggestion
You can use compare changes in development to preview data changes caused by your local edits (like added/removed rows and so on) directly in your editor, and without waiting on CI. Compare changes in development is available through the dbt VS Code extension, which is powered by the <Constant name="fusion_engine"/>.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+---
+title: "Compare changes locally"
+id: "vs-compare-changes"
+description: "Compare how code edits will change your data while you develop in locally with the dbt VS Code extension — without waiting on CI."
```
```suggestion
description: "Compare how code edits will change your data while you develop your dbt platform project locally with the dbt VS Code extension."
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+# Compare changes in local development <Lifecycle status="beta,managed,managed_plus" />
+
+:::info
+Compare changes in local development, powered by <Constant name="fusion_engine" />, is available in beta for the VS Code extension.
```
Do we need to do both the beta tag and the callout?

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+Compare changes in local development, powered by <Constant name="fusion_engine" />, is available in beta for the VS Code extension.
+:::
+
+The dbt VS Code extension, powered by the <Constant name="fusion_engine"/>, can preview and compare changes to your data caused by your local edits (like added/removed rows and so on) &mdash; directly in your editor and without waiting on CI. The dbt extension supports VS Code, Cursor, and Windsurf.
```
And so on is a little vague for the feature intro, what if we open with the description and then give a couple of examples. Stole these from futher down
```suggestion
The dbt VS Code extension previews and compares how your local edits affect your data in your <Constant name="dbt_platform"/> account &mdash;  including added/removed rows and join verification &mdash;  without waiting on CI.
- 
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+The dbt VS Code extension, powered by the <Constant name="fusion_engine"/>, can preview and compare changes to your data caused by your local edits (like added/removed rows and so on) &mdash; directly in your editor and without waiting on CI. The dbt extension supports VS Code, Cursor, and Windsurf.
+
+Use compare changes to check impact early and catch changes before you open a PR or run a [CI job](/docs/deploy/ci-jobs):
```
```suggestion
Use compare changes to check impact early and validate changes before you open a PR or run a [CI job](/docs/deploy/ci-jobs):
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+The dbt VS Code extension, powered by the <Constant name="fusion_engine"/>, can preview and compare changes to your data caused by your local edits (like added/removed rows and so on) &mdash; directly in your editor and without waiting on CI. The dbt extension supports VS Code, Cursor, and Windsurf.
+
+Use compare changes to check impact early and catch changes before you open a PR or run a [CI job](/docs/deploy/ci-jobs):
+- For example, when refactoring logic, adding or removing columns, or verifying a join modification doesn't change the data's output. 
```
```suggestion
- Validate outputs are correct when refactoring logic, adding or removing columns, or implementing join modifications. 
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- For example, when refactoring logic, adding or removing columns, or verifying a join modification doesn't change the data's output. 
+- It compares your current working copy against your `manifest.json` (for example, your last production state) and shows changes to primary keys, rows, and columns in the **Compare** tab. 
+
+This is [different from Advanced CI compare changes](#how-this-differs-from-advanced-ci), which runs at the PR stage in deployment rather than locally during development
```
It might be beneficial to put this at the top of the page in the opening so the user can get to the correct page earlier in their journey if they're in the wrong place

matthewshaver replied:
```suggestion
The dbt VS Code extension's compare changes feature is different from the [Advanced CI compare changes feature](#how-this-differs-from-advanced-ci), which runs at the PR stage in deployment rather than locally during development.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+This is [different from Advanced CI compare changes](#how-this-differs-from-advanced-ci), which runs at the PR stage in deployment rather than locally during development
+
+Compare changes in development is available for models only. Support for seeds, snapshots, ephemeral models, and Python models is coming soon.
```
This might be a good candidate for a callout

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+## Prerequisites
+
+- You have a dbt [Enterprise or Enterprise+](https://www.getdbt.com/pricing) account.
```
Is this the first major VSCE feature we're tying to platform account subscription? We should include this in the LSP features table and start putting the language out there that some features may require a paid account

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+<Lightbox src="/img/docs/extension/vs-compare-changes.gif" width="100%" title="Make changes to a model and see the changes in the Compare tab" />
+
+## Prerequisites
+
```
To use the dbt VS Code extension compare changes feature:

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+## Prerequisites
+
+- You have a dbt [Enterprise or Enterprise+](https://www.getdbt.com/pricing) account.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
```
```suggestion
- You've Installed the [dbt VS Code extension](/docs/install-dbt-extension) to your code editor and have a local installation of the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion).
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+- You have a dbt [Enterprise or Enterprise+](https://www.getdbt.com/pricing) account.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
```
```suggestion
- You've enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- You have a dbt [Enterprise or Enterprise+](https://www.getdbt.com/pricing) account.
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
+- Use a <Constant name="fusion" /> supported data platform: BigQuery, Databricks, Redshift, or Snowflake. Support for additional data platforms coming soon.
```
Might be helpful to link out to this so we don't have to come back and edit
```suggestion
- You use a <Constant name="fusion" /> [supported data platform](/docs/fusion/supported-features?version=2.0#requirements).
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- Installed the [dbt VS Code extension](/docs/install-dbt-extension) and the [<Constant name="fusion_engine" />](/docs/fusion/get-started-fusion) in your editor.
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
+- Use a <Constant name="fusion" /> supported data platform: BigQuery, Databricks, Redshift, or Snowflake. Support for additional data platforms coming soon.
+- A local `dbt_cloud.yml` file in your `.dbt` directory (for example, `~/.dbt/dbt_cloud.yml` on macOS and Linux). [Download](/docs/install-dbt-extension?version=2.0#register-with-dbt_cloudyml) it from your <Constant name="dbt_platform" /> account. The extension reads `dbt_cloud.yml` to authenticate. Without that file, compare changes cannot connect to <Constant name="dbt_platform" />.
```
```suggestion
- You have a `dbt_cloud.yml` file in your local `.dbt` directory (for example, `~/.dbt/dbt_cloud.yml` on macOS and Linux). [Download](/docs/install-dbt-extension?version=2.0#register-with-dbt_cloudyml) it from your <Constant name="dbt_platform" /> account. The extension reads `dbt_cloud.yml` to authenticate. Without that file, compare changes cannot connect to <Constant name="dbt_platform" />.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- Enabled [Advanced CI features](/docs/cloud/account-settings#enabling-advanced-ci-features) in your <Constant name="dbt_platform" /> account.
+- Use a <Constant name="fusion" /> supported data platform: BigQuery, Databricks, Redshift, or Snowflake. Support for additional data platforms coming soon.
+- A local `dbt_cloud.yml` file in your `.dbt` directory (for example, `~/.dbt/dbt_cloud.yml` on macOS and Linux). [Download](/docs/install-dbt-extension?version=2.0#register-with-dbt_cloudyml) it from your <Constant name="dbt_platform" /> account. The extension reads `dbt_cloud.yml` to authenticate. Without that file, compare changes cannot connect to <Constant name="dbt_platform" />.
+- If you've configured [automatic deferral](/docs/cloud/about-cloud-develop-defer), you need a successful job run. To use compare changes manually without a successful job run, copy the [`manifest.json` file](/reference/artifacts/manifest-json?version=2.0) into a folder and set the State directory in the [dbt extension settings](/docs/configure-dbt-extension#dbt-extension-settings) to that folder ...
```
```suggestion
- You have a successful job run if you've configured [automatic deferral](/docs/cloud/about-cloud-develop-defer). To use compare changes manually without a successful job run, copy the [`manifest.json` file](/reference/artifacts/manifest-json?version=2.0) into a folder and set the `State` directory in the [dbt extension settings](/docs/configure-dbt-extension#dbt-extension-settings) to that folder.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+#### How this differs from Advanced CI
+
+This compare changes feature applies to development only. If you're looking to compare changes between your production environment and the pull request’s latest commit, check out [Advanced CI compare changes](/docs/deploy/advanced-ci#compare-changes).
```
```suggestion
The dbt VS Code extension's compare changes feature applies only to your local development environment. If you're looking to compare changes between your production environment and the pull request’s latest commit, check out [Advanced CI compare changes](/docs/deploy/advanced-ci#compare-changes).
```

matthewshaver commented on website/static/img/docs/extension/vs-compare-changes.gif:
Might be helpful to edit out the load time in the GIF

matthewshaver commented on website/snippets/_compare-changes-tldr.md:
```diff
@@ -0,0 +1,7 @@
+## Compare changes in development <Lifecycle status="beta,managed,managed_plus" />
+
+You can use compare changes in development, powered by the <Constant name="fusion_engine"/>, to compare your current working copy against your `manifest.json` (for example, your last production state) — directly in your editor.
```
```suggestion
You can use compare changes, powered by the <Constant name="fusion_engine"/>,  in your local development environment to compare your current working copy against your `manifest.json` (for example, your last production state) directly in your editor.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+---
+title: "Compare changes locally"
```
Can we make the title a little more descriptive? If this is the first feature that the VS Code extension offers that really requires a platform subscription, this title may not have enough info. It might cause confusion sandwiched between other articles that don't have the same requirements. Maybe something like:
```suggestion
title: "Compare changes locally with your dbt platform account"
```

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Update website/docs/docs/deploy/advanced-ci.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+1. Open a SQL model file in your editor.
+2. Make some changes to the model, like adding a new column or modifying an existing one (for example, `left_join_customers` to `right_join_customers`).
+3. Run compare changes in the [VS Code interface](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) in the following ways:
```
```suggestion
3. Run compare changes in the [VS Code interface](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette):
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+   - Bottom panel: Click the **Compare** tab and then click the **Compare** button.
+  
+    <Lightbox src="/img/fusion/connect-adapters/vs-compare-changes.png" width="90%" title="Compare changes in development" />
+4. Once you click the **Compare** button, the extension will execute a `dbt build` command to build the model and then runs the comparison. The [**Compare** tab](#compare-tab) displays the changes to the data's primary keys, rows, and columns. Clicking the tabs will display more details about the changes, like specific columns that were added or modified.
```
Does it execute `dbt build` for everything, build only the model being viewed, or build all changed models?

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+## How it works
+
+Compare changes in development works by comparing two materialized models in your warehouse — it does not compare SQL file contents. Specifically, it compares the model built in your dev schema (determined by your active profile) against the model referenced in your `manifest.json` (for example, your last production state). Both sides of the comparison are always warehouse tables, not source file ...
```
```suggestion
Compare changes in development works by comparing two materialized models in your warehouse. Specifically, it compares the model built in your dev schema (determined by your active profile) against the model referenced in your `manifest.json` (for example, your last production state). Both sides of the comparison are always warehouse tables; it does not compare SQL file contents.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+- **Modified rows tab**: Details about the modified rows.
+- **Columns tab**: Details about the changes to the columns.
+
+## How it works
```
This section is great. It also reiterates some stuff already mentioned above in prerequisites. It could better serve the reader if we place it higher up (maybe right before ## Use compare changes) and remove some of the duplicate info around mentioning the deferral settings

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+
+
+<Expandable alt_header="Are queries run on behalf of the developer?"> 
+  Yes. All comparison queries in development are run using your local development credentials, directly from the dbt VS Code extension. The <Constant name="fusion_engine" /> uses your credentials to execute comparison queries in your warehouse. The results are stored in memory, so that we can keep them populated into the Compare tab for that file until you re-run.
```
Wouldn't this technically be their platform development credentials from the dbt_cloud.yml file?

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+title: "Compare changes locally"
+id: "vs-compare-changes"
+description: "Compare how code edits will change your data while you develop in locally with the dbt VS Code extension — without waiting on CI."
+sidebar_label: "Compare changes"
```
```suggestion
sidebar_label: "Compare changes locally"
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+</Expandable>
+<Expandable alt_header="Do I need to run `dbt build` every time I make a change?"> 
+  You don’t need to run `dbt build` after every change, but you do need at least one successful build per model to establish the baseline.
+  - You need one successful `build` of the model to create a baseline table for comparison.
```
One successful build anywhere or in an environment other than development?

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+  - If your model is invalid (for example, it has errors or breaking schema changes), dbt can't build it, so the comparison can't run until the model builds successfully.
+</Expandable>
+<Expandable alt_header="Does compare changes work with Python models?"> 
+  Not yet, but Python models are expected to work once Fusion support is available because compare changes only relies on the resulting table in the warehouse.
```
We've added support for Python models in Fusion!

mirnawong1 committed:
Apply suggestions from code review

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update compare changes documentation for Python models

Removed note about Python models not yet working with compare changes.

mirnawong1 committed:
Apply suggestions from code review

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Update advanced-ci.md

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,92 @@
+
+The dbt VS Code extension's compare changes feature is different from the [Advanced CI compare changes feature](#how-this-differs-from-advanced-ci), which runs at the PR stage in deployment rather than locally during development.
+
+Compare changes in development is available for models only. Support for seeds, snapshots, ephemeral models, and Python models is coming soon.
```
```suggestion
:::info
Compare changes in development is available for models only. Support for seeds, snapshots, ephemeral models, and Python models is coming soon.
:::
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,94 @@
+<Lightbox src="/img/docs/extension/vs-compare-changes.gif" width="100%" title="Make changes to a model and see the changes in the Compare tab" />
+
+## Prerequisites
+
```
```suggestion
To use the dbt VS Code extension compare changes feature:
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 committed:
matt feedback

mirnawong1 committed:
clarify baseline

mirnawong1 committed:
Update vs-compare-changes.md

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,98 @@
+# Compare changes in local development <Lifecycle status="beta,managed,managed_plus" />
+
+:::info
+This feature is in beta and rolling out to VS Code extension users on a <Constant name="dbt_platform" /> Enterprise or Enterprise+ account.
```
```suggestion
This feature is in beta and rolling out to VS Code extension users on a <Constant name="dbt_platform" /> Enterprise or Enterprise+ accounts.
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 committed:
Update compare changes documentation

Added a new section explaining how the compare changes feature works and removed redundant prerequisites section.

mirnawong1 commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,96 @@
+# Compare changes in local development <Lifecycle status="beta,managed,managed_plus" />
+
+:::info
+This feature is in beta and rolling out to VS Code extension users on a <Constant name="dbt_platform" /> Enterprise or Enterprise+ accounts.
```
```suggestion
This feature is in beta and rolling out to VS Code extension users on <Constant name="dbt_platform" /> Enterprise or Enterprise+ accounts.
```

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

mirnawong1 committed:
update images

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,99 @@
+# Compare changes in local development <Lifecycle status="beta,managed,managed_plus" />
+
+:::info
+This feature is in beta and rolling out to VS Code extension users on <Constant name="dbt_platform" /> Enterprise or Enterprise+ accounts.
```
```suggestion
This feature is in beta and rolling out to dbt VS Code extension users on <Constant name="dbt_platform" /> Enterprise or Enterprise+ accounts.
```

matthewshaver commented on website/docs/docs/fusion/vs-compare-changes.md:
```diff
@@ -0,0 +1,99 @@
+  - If you're using <Constant name="dbt_platform" />'s deferral (recommended): You need at least one successful job run in the environment you are deferring to (usually staging or production). This allows <Constant name="fusion" /> to auto-download the deferred manifest and use that as your baseline state to compare against.
+  - If you're manually setting a `state` directory: You can manually point the extension to a `manifest.json` (for example, copied from another environment) without needing a job run. <br />
+
+<Expandable alt_header="How is this different from Advanced CI compare changes?">
```
This could be more helpful down in the FAQs section

mirnawong1 committed:
Update website/docs/docs/fusion/vs-compare-changes.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'current' into vsce-compare-changes

mirnawong1 committed:
Refactor compare changes section in documentation

mirnawong1 committed:
Update vs-compare-changes.md
